From 13feb9e55fcf2b75da3ab9fd10c8dfa21db97c2b Mon Sep 17 00:00:00 2001
From: Erik Schumacher <Erik.Schumacher@iris-sensing.com>
Date: Thu, 24 Mar 2022 10:07:11 +0100
Subject: [PATCH 12/29] Hack: imx8-media-dev: Disable unregistering of mxc_isi
 media device

Bad approach! TODO: Distinguish probe failure <> deferred probing
This simply removes the unregister function in imx8-media-dev driver
Jira Task: https://jira.iris-sensing.net/browse/RDPHOEN-1059

Signed-off-by: Erik Schumacher <Erik.Schumacher@iris-sensing.com>
---
 drivers/staging/media/imx/imx8-media-dev.c | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/drivers/staging/media/imx/imx8-media-dev.c b/drivers/staging/media/imx/imx8-media-dev.c
index 4d019bf33a82..032a45e79f64 100644
--- a/drivers/staging/media/imx/imx8-media-dev.c
+++ b/drivers/staging/media/imx/imx8-media-dev.c
@@ -264,23 +264,6 @@ static int mxc_md_clean_unlink_channels(struct mxc_md *mxc_md)
 	return 0;
 }
 
-static void mxc_md_unregister_all(struct mxc_md *mxc_md)
-{
-	struct mxc_isi_info *mxc_isi;
-	int i;
-
-	for (i = 0; i < MXC_ISI_MAX_DEVS; i++) {
-		mxc_isi = &mxc_md->mxc_isi[i];
-		if (!mxc_isi->sd)
-			continue;
-
-		v4l2_device_unregister_subdev(mxc_isi->sd);
-		media_entity_cleanup(&mxc_isi->sd->entity);
-
-		pr_info("unregister ISI channel: %s\n", mxc_isi->sd->name);
-	}
-}
-
 static int mxc_md_create_links(struct mxc_md *mxc_md)
 {
 	struct media_entity *source, *sink;
@@ -1015,9 +998,6 @@ static int mxc_md_probe(struct platform_device *pdev)
 					goto clean_ents;
 
 				mxc_md_clean_unlink_channels(mxc_md);
-			} else {
-				/* no sensors connected */
-				mxc_md_unregister_all(mxc_md);
 			}
 		}
 	}
-- 
2.25.1

